File: src/main/java/guru/xgm/vjass/extapi/psi/VjassPsiFileBase.java
package guru.xgm.vjass.extapi.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import guru.xgm.vjass.lang.VjassLanguage;
import guru.xgm.vjass.openapi.fileTypes.VjassFileType;
import org.jetbrains.annotations.NotNull;

public class VjassPsiFileBase extends PsiFileBase {

    public VjassPsiFileBase(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, VjassLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return VjassFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "VJASS";
    }
}

File: src/main/kotlin/raft/war/language/vjass/extapi/psi/VjassPsiFileBase.kt
package raft.war.language.vjass.extapi.psi

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import raft.war.language.vjass.lang.VjassLanguage
import raft.war.language.vjass.VjassFileType

class VjassPsiFileBase(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, VjassLanguage.instance) {
    override fun getFileType(): FileType = VjassFileType.instance

    override fun toString(): String = VjassLanguage.NAME
}
